akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/LeastShardAllocationStrategy.scala

```diff
@@ -101,20 +103,25 @@ import akka.cluster.sharding.ShardRegion.ShardId
       // one rebalance at a time
       emptyRebalanceResult
     } else {
-      val numberOfShards = currentShardAllocations.valuesIterator.map(_.size).sum
-      val numberOfRegions = currentShardAllocations.size
-      if (numberOfRegions == 0 || numberOfShards == 0) {
+      val sortedRegionEntries = regionEntriesFor(currentShardAllocations)
```

Comment:
that is not sorted with the `ShardSuitabilityOrdering`?

Reply:
Good catch, it was intended to be sorted.

Reply:
Ups, added one sorting too many.

Reply:
Sorting now explicit to avoid surprises.